### PR TITLE
nginx: update nginx-dav-ext-module to 3.0

### PR DESCRIPTION
**Pull request by Ansuel**
Address request from #10195 

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

**Comment by cotequeiroz** [comment]
This needs run-testing.  @alex9434, since you requested this change, can you run-test it, please? Or has anybody else done that?

File: net/nginx/Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=nginx
 PKG_VERSION:=1.16.1
-PKG_RELEASE:=1
+PKG_RELEASE:=3
```
Comment (cotequeiroz) [inline net/nginx/Makefile]: Any reason to skip `2`?

Comment (Ansuel) [reply net/nginx/Makefile]: yes there is another pr to be merged before this

File: net/nginx/patches-dav-nginx/100-drop-libxslt-dep.patch [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ # be linked anyway.  In other cases libxslt is just redundant.
+ # If that's a big deal, libxml2 can be linked directly:
+-# ngx_module_libs=-lxml2
++ngx_module_libs=-lxml2
++ngx_module_incs=$(pkg-config --cflags-only-I libxml-2.0 | sed 's/^-I//')
+ 
+-ngx_module_libs=LIBXSLT
+-
+ ngx_module_srcs="$ngx_addon_dir/ngx_http_dav_ext_module.c"
+
```
Comment (yousong) [inline net/nginx/patches-dav-nginx/100-drop-libxslt-dep.patch]: Please add a newline at EOF.

**Comment by Ansuel** [comment]
Merge other before this

**Comment by alex9434** [comment]
Hi @Ansuel, I can confirm that this version compiled successfully.

**Comment by Ansuel** [comment]
Pls run test this

File: net/nginx/Makefile [hunk 3]
```diff
@@ -523,18 +523,19 @@ endif
 
 ifeq ($(CONFIG_NGINX_DAV),y)
   define Download/nginx-dav-ext-module
-    VERSION:=430fd774fe838a04f1a5defbf1dd571d42300cf9
+    VERSION:=f5e30888a256136d9c550bf1ada77d6ea78a48af
     SUBDIR:=nginx-dav-ext-module
     FILE:=nginx-dav-ext-module-$$(VERSION).tar.gz
```
Comment (dengqf6) [inline net/nginx/Makefile]: xz can be used to save space

File: net/nginx/Makefile [hunk 4]
```diff
@@ -523,18 +523,19 @@ endif
...
     URL:=https://github.com/arut/nginx-dav-ext-module.git
-    MIRROR_HASH:=0566053a8756423ecab455fd9d218cec1e017598fcbb3d6415a06f816851611e
+    MIRROR_HASH:=052dfc5a40b3b66ce27120e3ddab2a3dc9acbcb0184554e2b07552b2c490dc72
     PROTO:=git
   endef
   $(eval $(call Download,nginx-dav-ext-module))
 
   define Prepare/nginx-dav-ext-module
 	$(eval $(Download/nginx-dav-ext-module))
 	gzip -dc $(DL_DIR)/$(FILE) | tar -C $(PKG_BUILD_DIR) $(TAR_OPTIONS)
```
Comment (dengqf6) [inline net/nginx/Makefile]: Replace `gzip -dc` with `xzcat` if you use xz

File: net/nginx/Makefile (hunk 3 above)
Comment (Ansuel) [reply net/nginx/Makefile]: How to download xz from github?

Comment (dengqf6) [reply net/nginx/Makefile]: Just change gz to xz, the archive is created locally. See commit openwrt/packages@7d052e50123b5f54bfeb9e968b20e30e4a767730

**Comment by alex9434** [comment]
I would like to confirm that the nginx-dav-ext-module 3.0 from this pull request runs on my system (OpenWRT 18.06.4) without problems and according expectations.

**Comment by Ansuel** [comment]
@alex9434 ok then we can wait for you

**Comment by alex9434** [comment]
@Ansuel I apologize, but I do not understand. I have tested the changes and they work on my system. What are you waiting for? What should I do now? (unfortunately I am new to this process)

**Comment by Ansuel** [comment]
I thought you still had to test it... The u think this can be merged

Il Lun 28 Ott 2019, 06:31 Alexander <notifications@github.com> ha scritto:

**Pull request merged**